### DISCUSSION TITLE
Shipping Boost with a default HTTP load balancer

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/boost#750] [Ideas]
After discussing with PikNik in detail about the upcoming http retrieval feature, team pointed out that expecting people to have the technical ability to configure a load balance correctly might not be correct.
Learning and dealing with these tools, especially with the security scenarios might be too much for new SPs.

PikNik has advised us to document at least one default open-source tool (maybe nginix) and basic secure configuration to begin with. This would save time for new SPs who might not be familiar with load balancers and CDNs. It would also reduce the time to setup and start a boost http provider.
Another point raised was that by increasing the complexity of boost deployment, we might end up turning potential market users aways from using Boost.

**Comment by LexLuthr** [comment]
CC: @stuberman @nonsense @dirkmc @brendalee @honghao @LaurenSpiegel @jacobheun

**Comment by stuberman** [comment]
I want to reiterate, that many SP lack the experience to develop software or build systems (such as NGINIX) that are not included with a software package. SPs typically have infrastructure engineers that can configure systems but often lack software engineers. Ideally Lotus/Boost software will come with all software components ready to go with basic configuration only required. Asking for SPs to find and configure web authentication tools to protect Boost/booster-http from attacks is more than many will be prepared for. Ideally the booster-http component will include authentication mechanisms built in (whether peerID, wallet, minerID or other) such that only authorized Lotus/Boost clients can access these deals.

**Comment by LaurenSpiegel** [comment]
I thought we were going to document nginx usage? Having the http server turned on is optional and the immediate use case is evergreen. Is a load balancer even needed for this initial limited use case?

**Reply by brendalee** [reply]
Yes - I think the initial thinking is to document the basic option here for folks who would want to use this.
